Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalEnumerate.mm

```diff
@@ -41,26 +41,24 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 
 *==LICENSE==*/
 
-#include "HeadSpin.h"
-
 #include <vector>
 #include <string_theory/format>
-#include <Foundation/Foundation.hpp>
+#include <CoreGraphics/CoreGraphics.h>
+#include <Foundation/Foundation.h>
+#include <Metal/Metal.h>
 
 #include "plMetalPipeline.h"
 
 void plMetalEnumerate::Enumerate(std::vector<hsG3DDeviceRecord>& records)
 {
-    // For now - just use the default device. If there is a high power discrete device - this will
-    // spin it up. This will also automatically pin us to an eGPU if present and the user has
-    // configured us to use it.
-    MTL::Device* device = MTL::CreateSystemDefaultDevice();
+    CGDirectDisplayID mainDisplay = CGMainDisplayID();
+    id<MTLDevice> device = CGDirectDisplayCopyCurrentMetalDevice(mainDisplay);
```

Comment:
I'm actually suspicious of what this might do on a dual GPU MacBook Pro. This may not kick on the discrete GPU.

Plasma's original architecture definitely never had to contend with GPU switching. If this doesn't force a GPU switch I could force the GPU a different way.

I'll need to get out a dual GPU MacBook Pro to test this.

Reply:
This does cause the discrete GPU to spin up on a dual GPU MacBook Pro.